### PR TITLE
Build: refactoring grunt for clearer folder structure, tweaks

### PR DESCRIPTION
**Pull request by folletto**
This PR refactor the repository for clarity:

1. All source files are now in `sources/`, svg sources in `sources/svg/`
2. ~Adds PHP library build~ (see below)
3. Clarified gruntfile comments
4. All the builds are based on `svg-min` now instead of `svg`
5. Removed titles for accessibility reasons.

This aligns the build script to: https://github.com/Automattic/gridicons/pull/217

## To test

1. Clear the repository (ideally clone the repository again)
2. Try to reinstall / rebuild everything `npm install` + `npm run build`
3. Check the files built with this PR match the files before this PR

### CODE REVIEW

**Comment by ryelle** [comment]
When this was first ported over, the PHP section was left out, see my comment here: https://github.com/WordPress/dashicons/pull/169#issuecomment-293669228

**Comment by folletto** [comment]
It's fairly simple, so I'm ok either way. 

✅  Removed from this PR.

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,6 @@
 node_modules/
-svg/output/
```
Comment (jasmussen) [inline .gitignore]: Wondering why you removed `svg/output` and `build` directories from the git ignore?

Comment (folletto) [inline .gitignore]: Yep so:

* `build` is now simply `react`([ref](https://github.com/WordPress/dashicons/pull/197/files#diff-35b4a816e0441e6a375cd925af50752cR116))
* `svg/output` isn't used anywhere afaik, so just cleaned up

**Comment by jasmussen** [comment]
With 202 merged I suppose we can close this one right?

**Comment by folletto** [comment]
This was merged as #202. Closing.

**Pull request closed without merging**